code/game/machinery/machinery.dm

```diff
@@ -650,3 +650,19 @@ Class Procs:
 	. = . % 9
 	AM.pixel_x = -8 + ((.%3)*8)
 	AM.pixel_y = -8 + (round( . / 3)*8)
+
+/**
+ * Makes sure the user is allowed to interact with the machine when they use a shortcut, like Control or Alt-clicking.
+ *
+ * Arguments:
+ * * user - the mob who is trying to interact with the machine.
+ */
+/obj/machinery/proc/can_use_shortcut(mob/living/user)
+	if(!ishuman(user) && !issilicon(user))
+		return FALSE
+	if(!in_range(src, user) && !issilicon(user))
+		return FALSE
+	if(user.incapacitated())
```

Comment:
This instead maybe, to avoid two `issilicon()` checks?
```suggestion
/obj/machinery/proc/can_use_shortcut(mob/living/user)
	if(ishuman(user))
		if(!in_range(src, user))
			return FALSE
	else if(!issilicon(user))
		return FALSE
	if(user.incapacitated())
```

Reply:
Yeah my logic was crap here. I'm rewriting it a bit differently than you suggested though.
```dm
/obj/machinery/proc/can_use_shortcut(mob/living/user)
	if(user.incapacitated())
		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
		return FALSE
	if(ishuman(user) && in_range(src, user))
		return TRUE
	if(issilicon(user))
		return TRUE
	return FALSE

```